tests/test_vault.py

```diff
@@ -0,0 +1,185 @@
+        return MockResponse({'errors': ['Unknown user']}, status_code=401)
+
+    if (url.startswith('{}/v1/sys/leases/revoke'.format(host))):
+        lease_id = kwargs['json']['lease_id']
+        try:
+            leases.remove(lease_id)
+            return MockResponse(status_code=200)
+        except ValueError:
+            return MockResponse({'errors': ['Some other Error', 'NoSuchEntity', 'Error msg 2']}, 404)
+
+    return MockResponse(status_code=404)
+
+
+def mock_get(url, **kwargs):
+    if(url.startswith('{}/v1/{}/azure/creds/openqa-role'.format(host, namespace))):
+
+        lease_id = "azure/creds/openqa-role/{}".format(random_str(32))
+        leases.append(lease_id)
+        return MockResponse({
+            "lease_id": lease_id,
+            "lease_duration": 3600,
+            "data": {"client_id": random_str(32), "client_secret": random_str(32)},
+            })
+    elif(url.startswith('{}/v1/{}/secret/azure/openqa-role'.format(host, namespace))):
+        return MockResponse({
+            "data": {
+                "subscription_id": "XXXXX-5127-52311-31221-XXXXXXXXX",
+                "tenant_id": "XXXXXXX-ba41-4412-a321-XXXXXXXX"
+                }
+            })
+    elif(url.startswith('{}/v1/{}/aws/creds/openqa-role'.format(host, namespace))):
+        lease_id = "aws/creds/openqa-role/{}".format(random_str(32))
+        leases.append(lease_id)
+        return MockResponse({
+            "lease_id": lease_id,
+            "lease_duration": 3600,
+            "data": {
+                "access_key": random_str(32),
+                "secret_key": random_str(32),
+                "security_token": None
+              }
+            })
+
+    elif(url.startswith('{}/v1/{}/gcp/key/openqa-role'.format(host, namespace))):
+        keydata = """{{
+  "type": "service_account",
+  "project_id": "pcw",
+  "private_key_id": "{}",
+  "private_key": "{}",
+  "client_email": "vaultopenqa-role-XXXXXXX@pcw.iam.gserviceaccount.com",
+  "client_id": "{}",
+  "auth_uri": "https://accounts.google.com/o/oauth2/auth",
+  "token_uri": "https://oauth2.googleapis.com/token",
+  "auth_provider_x509_cert_url": "https://www.googleapis.com/oauth2/v1/certs",
+  "client_x509_cert_url": "https://www.googleapis.com/robot/v1/metadata/x509/vaultopenqa-role-XXXXXXX%40pcw.iam.gservi\
+ceaccount.com"
+}}""".format(random_str(32), random_str(32), random_str(32))
+
+        lease_id = "gcp/key/openqa-role/{}".format(random_str(32))
+        leases.append(lease_id)
+        return MockResponse({
+            "lease_id": lease_id,
+            "lease_duration": 3600,
+            "data": {
+                "key_algorithm": "KEY_ALG_RSA_2048",
+                "key_type": "TYPE_GOOGLE_CREDENTIALS_FILE",
+                "private_key_data": base64.b64encode(keydata.encode('UTF-8')).decode(encoding='UTF-8')
+              }
+            })
+
+
+def test_vault(monkeypatch):
```

Comment:
better to split this mega big test to dozen of one-liners where each one will cover some small piece of functionality